04 Research Environment/03 Datasets/08 Equity Options/10 Get Price Model Data.html

```diff
@@ -1,61 +1,166 @@
-<p>Follow these steps to get the values of theoretical prices, implied volatility, and Greeks:</p>
+<p>Follow these steps to get the values of implied volatility, and Greeks:</p>
 
 <ol>
-    <li><a href="/docs/v2/research-environment/datasets/equity-options#03-Create-Subscriptions">Create subscriptions and set the price model</a>.</li>
-    <li>Set the underlying <a href="/docs/v2/writing-algorithms/reality-modeling/options-models/volatility/key-concepts">volatility model</a>.</li>
+    <li><a href="/docs/v2/research-environment/datasets/equity-options#03-Create-Subscriptions">Create subscriptions</a>.</li>
     <div class="section-example-container">
-        <pre class="python">qb.Securities[equity_symbol].VolatilityModel = StandardDeviationOfReturnsVolatilityModel(30, Resolution.Daily)</pre>
+        <pre class="python">equity_symbol = qb.AddEquity("SPY", dataNormalizationMode=DataNormalizationMode.Raw).Symbol
+
+start_date = datetime(2021, 12, 31)
+contract_symbols = qb.OptionChainProvider.GetOptionContractList(equity_symbol, start_date)
+
+contract_symbol = [s for s in contract_symbols 
+    if s.ID.OptionRight == OptionRight.Call 
+        and s.ID.StrikePrice == 477 
+        and s.ID.Date == datetime(2022, 1, 21)][0]
+mirror_symbol = [s for s in contract_symbols 
+    if s.ID.OptionRight == OptionRight.Put 
+        and s.ID.StrikePrice == 477 
+        and s.ID.Date == datetime(2022, 1, 21)][0]
```

Comment:
I think this example should have more than two contracts, since people might want to see Greeks across multiple (20-40) strikes with the same expiry (front contracts).